src/components/Translation.tsx

```diff
@@ -19,34 +20,8 @@ const transform = {
 // Renders the translation string for the given translation key `id`. It
 // fallback to English if it doesn't find the given key in the current language
 const Translation = ({ id, options }: Props) => {
-  const [requiredNamespaces, setRequiredNamespaces] = useState<string[]>([])
   const { asPath } = useRouter()
-
-  useEffect(() => {
```

Comment:
I've also removed this `useEffect` as it was not needed, and was causing a loop. `setState` => `useEffect dep triggered` => `setState again`